daemons/controld/controld_remote_ra.c

```diff
@@ -365,18 +349,15 @@ remote_node_down(const char *node_name, const enum down_opts opts)
     int call_opt = crmd_cib_smart_opt();
     crm_node_t *node;
 
-    /* Purge node from attrd's memory */
+    // Purge node's transient attributes (from attribute manager and CIB)
     update_attrd_remote_node_removed(node_name, NULL);
 
-    /* Normally, only node attributes should be erased, and the resource history
-     * should be kept until the node comes back up. However, after a successful
-     * fence, we want to clear the history as well, so we don't think resources
-     * are still running on the node.
+    /* Normally, the resource history should be kept until the node comes back
+     * up. However, after a successful fence, clear the history so we don't
+     * think resources are still running on the node.
      */
     if (opts == DOWN_ERASE_LRM) {
-        controld_delete_node_state(node_name, controld_section_all, call_opt);
-    } else {
-        controld_delete_node_state(node_name, controld_section_attrs, call_opt);
+        controld_delete_node_state(node_name, controld_section_lrm, call_opt);
```

Comment:
I *think* this patch won't break anything around my previous feature to preserve remote node transient attributes (09058c9439527ac55290484c190e82d35f14cb71 which unfortunately references the wrong upstream project), but I'm wondering if you were able to test that.

Reply:
Haven't done any testing yet, but I did have that in mind. Basically the only difference is that the direct clearing of attributes is replaced with an attrd IPC request.